### PR TITLE
Make error message for certs not found in user table clearer

### LINKED ISSUE #525: Error message for unknown cert is confusing
I accidently used the member cert for userrpc and got back "[INVALID_CALLER_ID]: No corresponding caller entry exists." This doesn't mean anything to me. How about something like.. "Could not find matching user certificate." ?

### PR DESCRIPTION
**Pull request by achamayou**
fix #525

### CODE REVIEW

File: src/node/rpc/frontend.h [hunk 1]
```diff
@@ -503,7 +503,7 @@ namespace ccf
           jsonrpc::error_response(
             0,
             jsonrpc::CCFErrorCodes::INVALID_CALLER_ID,
-            "No corresponding caller entry exists."),
+            "Could not find matching user certificate."),
```
Comment (eddyashton) [inline src/node/rpc/frontend.h]: This is also called by members, where this error is incorrect. We could use `callers->get_name()`, but I think that will be `ccf.user_certs`?

**Comment by cimetrics** [comment]
images

File: src/node/rpc/frontend.h (hunk 1 above)
Comment (achamayou) [reply src/node/rpc/frontend.h]: actor certificate?

Comment (eddyashton) [reply src/node/rpc/frontend.h]: Maybe overkill, but as a general solution:

`frontend.h`:
```
virtual std::string invalid_caller_error_message() const {
  return "Could not find matching actor certificate";
}
```

`memberfrontend.h`:
```
std::string invalid_caller_error_message() const override {
  return "Could not find matching member certificate";
}
```

etc

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**